eng/Generate.ps1

```diff
@@ -64,11 +64,17 @@ function Add-TestServer-Swagger ([string]$testName, [string]$projectSuffix, [str
 
 function Add-CadlRanch-TypeSpec([string]$testName, [string]$projectPrefix, [string]$cadlRanchProjectsDirectory) {
     $projectDirectory = Join-Path $cadlRanchProjectsDirectory $testName
+    $configFile = Join-Path $projectDirectory "tspconfig.yaml"
     if (Test-Path "$projectDirectory/*.sln") {
         $projectDirectory = Join-Path $projectDirectory "src"
     }
     $tspMain = Join-Path $cadlRanchFilePath $testName "main.tsp"
-    Add-TypeSpec "$projectPrefix$testName" $projectDirectory $tspMain
+    if (Test-Path $configFile) {
+        Add-Typespec "$projectPrefix$testName" $projectDirectory $tspMain "--config=$configFile"
```

Comment:
why is this needed, doesn't tsp look for the tspconfig.yaml in the same spot so it should automatically find this?

Reply:
@m-nash Some test projects have separate configuration, like https://github.com/Azure/autorest.csharp/blob/feature/v3/test/CadlRanchProjects/authentication/api-key/tspconfig.yaml

Those configuration files are not saved along with the typespec files. In `0.45.0`, there is a [breaking change](https://github.com/microsoft/typespec/pull/1942) on how typespec cli resolves the path of configuration file. So, we need to explicitly specify the path to the configuration file in our cases.

Reply:
One more thing, typespec team thinks it is a bug fix, so it is not in the breaking change section of release notes. I don't really agree with them.

Reply:
yeah, bug fixes can be breaking changes as well @pshao25.  @allenjzhang does tsp team consider bug fixes that are breaking changes as well should go into the breaking change section?

Reply:
@archerzz  is this because the tsp is coming from cadl-ranch so the tspconfig.yaml is not next to the main.tsp?

Reply:
@m-nash That's correct. All `tspconfig.yaml` files under https://github.com/Azure/autorest.csharp/tree/feature/v3/test/CadlRanchProjects are checked in our repo, not `cadl-ranch`.